### PR TITLE
Add GitHub actions

### PR DESCRIPTION
**Pull request by luck3y**
Signed-off-by: Ken Wills <kwills@redhat.com>

### CODE REVIEW

**Comment by luck3y** [comment]
@dmlloyd Sorry if these are pestering you, I just have an irresistible urge to move them out of TC.

This one was a bit of a puzzle based on the need for JDK8 + whatever else is being tested due to windows env vars and powershell.

Oh, and also, since master wasn't currently building, I opened this against 1.x, LMK if it should be master first.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+name: JBoss Modules CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  build-test-matrix:
+    name: JDK${{ matrix.jdk }}-${{ matrix.impl}}-${{ matrix.os }}
```
Comment (dmlloyd) [inline .github/workflows/ci.yml:12]: This project is using the jboss-parent MR JAR mechanism so it should not be built using matrix.  There aren't any instructions for setting up GH actions for this on the JBoss Parent project, but it's basically the same process as it is for SmallRye Parent which is documented here: https://github.com/smallrye/smallrye-parent/#using-mr-jar-functions-with-github-actions.....

Comment (dmlloyd) [reply .github/workflows/ci.yml:12]: Which I see you've essentially already done... let me read further.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      if: contains(matrix.os, 'windows')
+      run: |
+        echo "::set-env name=JAVA8_HOME::$env:JAVA_HOME"
+    - name: Set up JDK ${{ matrix.jdk }}
+      uses: AdoptOpenJDK/install-jdk@v1
+      with:
+        version: ${{ matrix.jdk }}
+        impl: ${{ matrix.impl }}
+    - name: Run Build & Test
+    # -D needs to be escaped in powershell, use --define= instead
```
Comment (dmlloyd) [inline .github/workflows/ci.yml:42]: :man_facepalming:

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    # test runs use -Djava8.home=... as well, so we need both JDKs
+    - name: Set up JDK 8
+      uses: AdoptOpenJDK/install-jdk@v1
+      with:
+        version: 8
+        impl: ${{ matrix.impl }}
+    - name: Store JDK8 env (Linux)
+      if: contains(matrix.os, 'ubuntu')
+      run: |
+        echo "::set-env name=JAVA8_HOME::$(echo $JAVA_HOME)"
```
Comment (dmlloyd) [inline .github/workflows/ci.yml:31]: Why bother with setting an env var here?  The AdoptOpenJDK install job should set `JAVA_HOME_8` for you already.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      run: |
+        echo "::set-env name=JAVA8_HOME::$env:JAVA_HOME"
+    - name: Set up JDK ${{ matrix.jdk }}
+      uses: AdoptOpenJDK/install-jdk@v1
+      with:
+        version: ${{ matrix.jdk }}
+        impl: ${{ matrix.impl }}
+    - name: Run Build & Test
+    # -D needs to be escaped in powershell, use --define= instead
+      run: mvn -U -B -fae clean install --define=java8.home=${{ env.JAVA8_HOME }}
```
Comment (dmlloyd) [inline .github/workflows/ci.yml:43]:
```suggestion
      run: mvn -U -B -fae clean install --define=java8.home=${{ env.JAVA_HOME_8 }}
```

Assuming it works on Windows that is.  If so, you could delete the "Store JDK8 env" bits...

Comment (dmlloyd) [reply .github/workflows/ci.yml:43]: It would be good to also add java9 and java10 TBH.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (luck3y) [reply .github/workflows/ci.yml:31]: Well, yeah it does, then installing openjdk 11 overwrites it ;)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (luck3y) [reply .github/workflows/ci.yml:12]: Oh, I see the TARGETs thing, I'll give that a go, thanks for the pointer.

**Comment by dmlloyd** [comment]
Yeah `master` is EOL; the default branch is `1.x`.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (luck3y) [reply .github/workflows/ci.yml:43]: +1 on the var rename, and yeah will add 9 & 10 as well

**Comment by dmlloyd** [comment]
I mean you don't have to *close* it :)

**Comment by luck3y** [comment]
Oh, I just closed it to open a test PR in my fork and make sure it worked ;)

**Comment by luck3y** [comment]
I guess I'll have to reopen as a different PR, odd, thats usually fine even if I force push..

**Pull request closed without merging**